src/ccache.cpp

```diff
@@ -789,18 +789,24 @@ static bool
 write_result(Context& ctx,
              const std::string& result_path,
              const Stat& obj_stat,
+             const std::string& stdout_path,
              const std::string& stderr_path)
 {
   Result::Writer result_writer(ctx, result_path);
 
+  const auto stdout_stat = Stat::stat(stdout_path, Stat::OnError::log);
   const auto stderr_stat = Stat::stat(stderr_path, Stat::OnError::log);
-  if (!stderr_stat) {
+  if (!stdout_stat || !stderr_stat) {
     return false;
   }
 
   if (stderr_stat.size() > 0) {
     result_writer.write(Result::FileType::stderr_output, stderr_path);
   }
+  // Write stdout only after stderr (better with MSVC).
```

Comment:
Why is the order important? Can you explain that a little bit more?

Reply:
MSVC output is a bit of a mess. Error messages go to stdout, and the banner that can be disabled with -nologo goes to stderr. So in the logical order you'd get the banner at the end.

Reply:
I took another look at it: 
Isn't that where it is written to the disk? 
The order doesn't matter, because they are two different files. 
I think it's better if you remove the comment here because it is more confusing.

Reply:
ResultRetriever processes items in the order it reads them from the disk.

Reply:
> ResultRetriever processes items in the order it reads them from the disk.

Ok then I propose: 
```suggestion
  // Write stdout only after stderr (better with MSVC), because the ResultRetriever
  // processes the items in the order it reads them from disk afterwards. 
```